django/db/models/query.py

```diff
@@ -560,16 +560,19 @@ def last(self):
             return objects[0]
         return None
 
-    def in_bulk(self, id_list=None):
+    def in_bulk(self, id_list=None, field='pk'):
         """
         Return a dictionary mapping each of the given IDs to the object with
         that ID. If `id_list` isn't provided, evaluate the entire QuerySet.
         """
         assert self.query.can_filter(), \
             "Cannot use 'limit' or 'offset' with in_bulk"
+        if field != 'pk' and not self.model._meta.get_field(field).unique:
+            raise ValueError("'field' argument must be unique")
```

Comment:
`django.core.exceptions.FieldDoesNotExist: Article has no field named 'whatever'`

And traceback points to the code I wrote to test it, so seems good to me?

Will change the wording on that `ValueError` though, it could be more explicit.